src/test/java/de/hpi/bpt/argos/api/EntityEndpointTest.java

```diff
@@ -0,0 +1,273 @@
+	}
+
+	@Test
+	public void testGetEntity() {
+		RestRequest request = RestRequestFactoryImpl.getInstance().createGetRequest(ARGOS_REST_HOST, getEntityUri(testEntity.getId()));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		JsonObject entity = jsonParser.parse(request.getResponse()).getAsJsonObject();
+		JsonArray entityAttributes = entity.get("Attributes").getAsJsonArray();
+
+		assertEquals(testEntity.getId(), entity.get("Id").getAsLong());
+		assertEquals(testEntity.getTypeId(), entity.get("TypeId").getAsLong());
+		assertEquals(testEntity.getParentId(), entity.get("ParentId").getAsLong());
+		assertEquals(testEntity.getName(), entity.get("Name").getAsString());
+		assertEquals(testEntity.getStatus(), entity.get("Status").getAsString());
+		assertEquals(testEntityAttributes.size(), entityAttributes.size());
+	}
+
+	@Test
+	public void testGetEntity_InvalidId_NotFound() {
+		RestRequest request = RestRequestFactoryImpl.getInstance().createGetRequest(ARGOS_REST_HOST, getEntityUri(testEntity.getId() - 1));
+
+		assertEquals(HttpStatusCodes.NOT_FOUND, request.getResponseCode());
+	}
+
+	@Test
+	public void testGetChildEntities() {
+		List<Object> attributeNames = new ArrayList<>();
+
+		for (int i = 0; i < 2; i++) {
+			attributeNames.add(testEntityTypeAttributes.get(i).getName());
+		}
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getChildEntitiesUri(testEntity.getParentId(), testEntity.getTypeId(), attributeNames));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		JsonArray childEntities = jsonParser.parse(request.getResponse()).getAsJsonArray();
+		assertEquals(1, childEntities.size());
+
+		JsonObject entity = childEntities.get(0).getAsJsonObject();
+		JsonArray entityAttributes = entity.get("Attributes").getAsJsonArray();
+
+		assertEquals(testEntity.getId(), entity.get("Id").getAsLong());
+		assertEquals(testEntity.getTypeId(), entity.get("TypeId").getAsLong());
+		assertEquals(testEntity.getParentId(), entity.get("ParentId").getAsLong());
+		assertEquals(testEntity.getName(), entity.get("Name").getAsString());
+		assertEquals(testEntity.getStatus(), entity.get("Status").getAsString());
+		assertEquals(attributeNames.size(), entityAttributes.size());
+	}
+
+	@Test
+	public void testGetChildEntities_InvalidParentId_Success() {
+		List<Object> attributeNames = new ArrayList<>();
+
+		for (int i = 0; i < 2; i++) {
+			attributeNames.add(testEntityTypeAttributes.get(i).getName());
+		}
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createGetRequest(ARGOS_REST_HOST, getChildEntitiesUri(testEntity.getParentId() - 1, testEntity.getTypeId(), attributeNames));
+
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		JsonArray childEntities = jsonParser.parse(request.getResponse()).getAsJsonArray();
+		assertEquals(0, childEntities.size());
```

Comment:
The `ParentId` is invalid in this test. This means, that there is no `Entity` with the given `ParentId` stored in the database. Therefore there must be no children whatsoever.